hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/FullScanPhysicalRule.java

```diff
@@ -46,4 +57,40 @@ public RelNode convert(RelNode rel) {
                 logicalScan.getTable()
         );
     }
+
+    @Override
+    public void onMatch(RelOptRuleCall call) {
+        RelNode rel = call.rel(0);
+        Table tableUnwrapped = getTableUnwrapped(rel.getTable());
+        List<RelNode> transforms = new ArrayList<>();
+
+        // Only PartitionedMapTable supposed to have indices.
+        if (tableUnwrapped instanceof PartitionedMapTable) {
```

Comment:
What about creating separate `IMap`-only rule that would be creating index rels?